homeassistant/components/samsungtv/media_player.py

```diff
@@ -246,10 +248,19 @@ async def async_play_media(self, media_type, media_id, **kwargs):
             await asyncio.sleep(KEY_PRESS_TIMEOUT, self.hass.loop)
         await self.hass.async_add_executor_job(self.send_key, "KEY_ENTER")
 
+    def _wake_on_lan(self):
+        """Wake the device via wake on lan."""
+        send_magic_packet(self._mac, ip_address=self._host)
+        # If the ip address changed since we last saw the device
+        # broadcast a packet as well
+        send_magic_packet(self._mac)
+
     async def async_turn_on(self):
         """Turn the media player on."""
         if self._on_script:
             await self._on_script.async_run(context=self._context)
+        elif self._mac:
+            await self.hass.async_add_executor_job(self._wake_on_lan)
```

Comment:
Note that we should not allow adding any options for this wake on lan usage. If users need options they will need to use the script.